### ISSUE TITLE
In the v2 version, when I call the open method to open a file directory, the program freezes[bug] 

### ISSUE THREAD

**Issue by pdap** [tauri-apps/tauri#9498]
### Describe the bug

In the v2 version, when I call the open method to open a file directory, the program freezes

### Reproduction

````js
import { open } from "@tauri-apps/plugin-dialog";
  open({
      directory:true
  }).then((f)=>{
      console.log(f);
  })
````

### Expected behavior

Return to the directory selected by the user

### Full `tauri info` output

```text
> um-sdk-tool@0.0.0 tauri /Users/pdop/gitdir/um-sdk-tool
> tauri "info"


[✔] Environment
    - OS: Mac OS 14.3.1 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.77.2 (25ef9e3d8 2024-04-09)
    ✔ cargo: 1.77.2 (e52e36006 2024-03-26)
    ✔ rustup: 1.27.0 (bbb9276d2 2024-03-08)
    ✔ Rust toolchain: stable-aarch64-apple-darwin (default)
    - node: 20.11.1
    - pnpm: 8.15.3
    - yarn: 1.22.22
    - npm: 10.4.0

[-] Packages
    - tauri [RUST]: 2.0.0-beta.15
    - tauri-build [RUST]: 2.0.0-beta.12
    - wry [RUST]: 0.39.1
    - tao [RUST]: 0.27.0
    - @tauri-apps/api [NPM]: 2.0.0-beta.8
    - @tauri-apps/cli [NPM]: 2.0.0-beta.13

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:1420/
    - framework: Vue.js
    - bundler: Vite
```

**Comment by FabianLars** [comment]
Thanks for the report! It works on my end just fine, could you maybe share a minimal reproduction repo we can use to test it?

**Comment by pdap** [comment]
> Thanks for the report! It works on my end just fine, could you maybe share a minimal reproduction repo we can use to test it?

You can reproduce it in this way. When you click the button to call the open method and then select a directory, there is no return value. If you click the select file button again, the program will freeze.

**Comment by FabianLars** [comment]
Here's what i am doing using the create-tauri-app react template and it works fine:
<img width="759" alt="grafik" src="https://github.com/tauri-apps/tauri/assets/30730186/b88b0652-607b-4246-88ed-80c7df99b21b">

**Comment by pdap** [comment]
> Here's what i am doing using the create-tauri-app react template and it works fine: <img alt="grafik" width="759" src="https://private-user-images.githubusercontent.com/30730186/323601458-b88b0652-607b-4246-88ed-80c7df99b21b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM0NDQxOTAsIm5iZiI6MTcxMzQ0Mzg5MCwicGF0aCI6Ii8zMDczMDE4Ni8zMjM2MDE0NTgtYjg4YjA2NTItNjA3Yi00MjQ2LTg4ZWQtODBjN2RmOTliMjFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE4VDEyMzgxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlMzlhY2IzMmEzNjYyZTUzZTYzYzNmYzZjMGEwMTI4MWJiODA0N2ZkZGJlM2MxOWY3OWNhNzEyOGZiYjcyZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vaN6i43SWbIsRMxc3mg3ca4UNNxZHVXrBQi-GQ2OpeM">

 I found that it is indeed good when using the open plug-in independently, but when introducing 'tauri-plugin-persisted-scope', and '@tauri-apps/plugin-fs', it will cause this problem
[This is a minimally reproducible demo](https://github.com/pdap/tauri-demo)

**Comment by FabianLars** [comment]
Thank you, i will look into it right now :)

**Comment by FabianLars** [comment]
Thanks again for the reproduction, this should be fixed by https://github.com/tauri-apps/plugins-workspace/pull/1221 - you can try this by adding this to your cargo.toml file and then running `cargo update` inside the src-tauri folder
```toml
[patch.crates-io]
tauri-plugin-fs = { git = "https://github.com/tauri-apps/plugins-workspace", branch = "fix/fs/deadlock" }
```

**Comment by pdap** [comment]
> Thanks again for the reproduction, this should be fixed by [tauri-apps/plugins-workspace#1221](https://github.com/tauri-apps/plugins-workspace/pull/1221) - you can try this by adding this to your cargo.toml file and then running `cargo update` inside the src-tauri folder
> 
> ```toml
> [patch.crates-io]
> tauri-plugin-fs = { git = "https://github.com/tauri-apps/plugins-workspace", branch = "fix/fs/deadlock" }
> ```

Thank you, the problem has been solved after applying this patch. :)
